tracee-rules/output_test.go

```diff
@@ -82,3 +91,101 @@ Hostname: foobar.local
 	}
 }
+
+func Test_sendToWebhook(t *testing.T) {
+	var testCases = []struct {
+		name               string
+		inputTemplateFile  string
+		inputSignature     fakeSignature
+		inputTestServerURL string
+		expectedOutput     string
+		expectedError      string
+	}{
+		{
+			name:           "happy path, no template JSON output",
+			expectedOutput: `{"output":"Rule \"foo bar signature\" detection:\n map[foo1:bar1, baz1 foo2:[bar2 baz2]]","rule":"foo bar signature","time":"2021-02-23T01:54:57Z","output_fields":{"value":0}}`,
+		},
+		{
+			name: "happy path, with simple template",
+			expectedOutput: `*** Detection ***
+Timestamp: 2021-02-27T07:19:07Z
+ProcessName: foobar.exe
+HostName: foobar.local
+`,
+			inputTemplateFile: "templates/simple.tmpl",
+		},
+		{
+			name:              "happy path, with CSV template",
+			expectedOutput:    `2021-02-27T07:19:07Z,foobar.exe,foobar.local`,
+			inputTemplateFile: "templates/csv.tmpl",
+		},
+		{
+			name: "happy path, with XML template",
+			expectedOutput: `<?xml version="1.0" encoding="UTF-8" ?>
+ <detection timestamp="2021-02-27T07:19:07Z">
+    <processname>foobar.exe</processname>
+    <hostname>foobar.local</hostname>
+ </detection>`,
+			inputTemplateFile: "templates/xml.tmpl",
+		},
+		{
+			name: "sad path, with failing GetMetadata func for sig",
+			inputSignature: fakeSignature{
+				getMetadata: func() (types.SignatureMetadata, error) {
+					return types.SignatureMetadata{}, errors.New("getMetadata failed")
+				},
+			},
+			expectedError: "error preparing json payload: getMetadata failed",
+		},
+		{
+			name:               "sad path, error reaching webhook",
+			inputTestServerURL: "foo://bad.host",
+			expectedError:      `error calling webhook Post "foo://bad.host": unsupported protocol scheme "foo"`,
+		},
+		{
+			name:              "sad path, with missing template",
+			inputTemplateFile: "invalid/template",
+			expectedError:     `error preparing webhook template: open invalid/template: no such file or directory`,
+		},
+		{
+			name:              "sad path, with an invalid template",
+			inputTemplateFile: "goldens/broken.tmpl",
+			expectedError:     `error writing to the template: template: broken.tmpl:1:3: executing "broken.tmpl" at <.InvalidField>: can't evaluate field InvalidField in type types.Finding`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				got, _ := ioutil.ReadAll(request.Body)
+				assert.Equal(t, tc.expectedOutput, string(got), tc.name)
+			}))
+			defer ts.Close()
+
+			if tc.inputTestServerURL != "" {
+				ts.URL = tc.inputTestServerURL
+			}
+
+			actualError := sendToWebhook(types.Finding{
+				Data: map[string]interface{}{
+					"foo1": "bar1, baz1",
+					"foo2": []string{"bar2", "baz2"},
+				},
+				Context: external.Event{
+					Timestamp:   1614410347,
```

Comment:
The context event's TS shouldn't be used as the TS of the detection, the system current TS should be (like you correctly did in the previous PR that formatted the stdout). 
(FYI, Tracee Event's TS is not a unix TS but that doesn't matter if you remove this info here)

Reply:
So your suggestion is to remove this and the template parsing logic to instead do a time.Now() as the value each time the template payload is generated? (rather than using this timestamp field)

Reply:
yes. if you notice, this is what you already did when you printed the detection to stdout. this timestamp is meaningless as a unix time value

Reply:
OK I got it.